src/main/resources/templates/smallfull/addOrRemoveLoans.html

```diff
@@ -26,241 +30,269 @@ <h2 id="loansToDirectors-subheading" class="govuk-heading-m">
+                                                    th:text="'Balance at ' + ${#temporals.format(addOrRemoveLoans.nextAccount.periodEndOn, 'd MMMM yyyy')} + ' in £'"></strong>
+                                        </div>
+                                        <div class="govuk-grid-column-one-half column-flex govuk-summary__align--right">
+                                            <strong th:id="loans-table-breakdown-balance-at-period-end[__${stat.index}__]"
+                                                    th:text="${loan.breakdown.balanceAtPeriodEnd}">
+                                            </strong>
                                         </div>
                                     </div>
-                                </details>
-                            </td>
-                        </tr>
-                    </th:block>
+                                </div>
+                            </details>
+                        </td>
+                    </tr>
+                </th:block>
                 </tbody>
             </table>
 
-            <form id="data-submission-form" th:action="@{''}" class="form currency-included-within-inputs govuk-body" th:object="${addOrRemoveLoans}" method="post">
+            <form id="data-submission-form" th:action="@{''}" class="form currency-included-within-inputs govuk-body"
+                  th:object="${addOrRemoveLoans}" method="post">
+
+                <input th:field="*{nextAccount.periodStartOn}" type="hidden">
+                <input th:field="*{nextAccount.periodEndOn}" type="hidden">
 
                 <div th:each="loan, stat : ${addOrRemoveLoans.existingLoans}">
-                    <input type="hidden" th:field="${addOrRemoveLoans.existingLoans[__${stat.index}__].directorName}"
+                    <input type="hidden" th:field="*{existingLoans[__${stat.index}__].directorName}"
                            th:id="existingLoans[__${stat.index}__].directorName"/>
-                    <input type="hidden" th:field="${addOrRemoveLoans.existingLoans[__${stat.index}__].breakdown.balanceAtPeriodEnd}"
+                    <input type="hidden"
+                           th:field="*{existingLoans[__${stat.index}__].breakdown.balanceAtPeriodEnd}"
                            th:id="existingLoans[__${stat.index}__].breakdown.balanceAtPeriodEnd"/>
-                    <input type="hidden" th:field="${addOrRemoveLoans.existingLoans[__${stat.index}__].id}"
+                    <input type="hidden" th:field="*{existingLoans[__${stat.index}__].id}"
                            th:id="existingLoans[__${stat.index}__].id"/>
                 </div>
 
-                <div class="govuk-form-group govuk-form-group-block-important">
-                      <div>
+                <div th:if="${addOrRemoveLoans.existingLoans == null or addOrRemoveLoans.existingLoans.length < 20}"
```

Comment:
We'll have to be careful on the merge ordering as you've pulled the logic from [518](https://github.com/companieshouse/company-accounts.web.ch.gov.uk/pull/518) into this PR